### PR TITLE
plugins.linelive: new plugin for LINE LIVE (live.line.me)

### PR DESCRIPTION
**Pull request by ssaqua**
#2573

### CODE REVIEW

File: src/streamlink/plugins/linelive.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        channel = match.group("channel")
+        broadcast = match.group("broadcast")
+        res = self.session.http.get(API_URL.format(channel, broadcast))
+        json = self.session.http.json(res, schema=_player_status_schema)
+        if json["liveStatus"] != "LIVE":
+            return
+        for stream in json["liveHLSURLs"]:
+            url = json["liveHLSURLs"][stream]
+            if url != None:
+                yield stream, HLSStream(self.session, url)
```
Comment (back-to) [inline src/streamlink/plugins/linelive.py]: use a stream name with `p` , so you don't need `stream_weight`

```py
"{0}p".format(stream)
```

File: src/streamlink/plugins/linelive.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    }
+)
+
+class LineLive(Plugin):
+    @classmethod
+    def can_handle_url(cls, url):
+        return _url_re.match(url) is not None
+
+    @classmethod
+    def stream_weight(cls, stream):
```
Comment (back-to) [inline src/streamlink/plugins/linelive.py]: not needed, can be removed

File: src/streamlink/plugins/linelive.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
+import re
+
+from streamlink.plugin import Plugin
+from streamlink.plugin.api import validate
+from streamlink.stream import HLSStream
+
+API_URL = "https://live-api.line-apps.com/app/v3.2/channel/{0}/broadcast/{1}/player_status"
+
+QUALITY_WEIGHTS = {
```
Comment (back-to) [inline src/streamlink/plugins/linelive.py]: not needed, can be removed

File: src/streamlink/plugins/linelive.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    "144": 144,
+}
+
+_url_re = re.compile(r"""
+    https?://live.line.me
+    /channels/(?P<channel>\d+)
+    /broadcast/(?P<broadcast>\d+)
+""", re.VERBOSE)
+
+_player_status_schema = validate.Schema(
```
Comment (back-to) [inline src/streamlink/plugins/linelive.py]: put this into `class LineLive`

File: src/streamlink/plugins/linelive.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+QUALITY_WEIGHTS = {
+    "720": 720,
+    "480": 480,
+    "360": 360,
+    "240": 240,
+    "144": 144,
+}
+
+_url_re = re.compile(r"""
```
Comment (back-to) [inline src/streamlink/plugins/linelive.py]: put this into `class LineLive`

File: src/streamlink/plugins/linelive.py [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    https?://live.line.me
```
Comment (back-to) [inline src/streamlink/plugins/linelive.py]: escape the dots `\.`

File: src/streamlink/plugins/linelive.py [hunk 7]
```diff
@@ -0,0 +1,63 @@
+import re
+
+from streamlink.plugin import Plugin
+from streamlink.plugin.api import validate
+from streamlink.stream import HLSStream
+
+API_URL = "https://live-api.line-apps.com/app/v3.2/channel/{0}/broadcast/{1}/player_status"
```
Comment (back-to) [inline src/streamlink/plugins/linelive.py]: put this into `class LineLive`

**Comment by ssaqua** [comment]
Thanks for reviewing @back-to, update made

**Pull request merged**